src/components/BookList.js

```diff
@@ -1,23 +1,43 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
+import { React, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import NewBookForm from './NewBookForm';
 import Book from './Book';
+import {
+  getBooksErrors, getBooksStatus, selectAllBooks, fetchBooks,
+} from '../redux/books/booksSlice';
 
 const BookList = () => {
-  const books = useSelector((store) => store.books.books);
+  const dispatch = useDispatch();
+  const books = useSelector(selectAllBooks);
+  const booksStatus = useSelector(getBooksStatus);
+  const booksErrors = useSelector(getBooksErrors);
+
+  useEffect(() => {
+    dispatch(fetchBooks());
+  }, [dispatch]);
+
+  let content;
+
+  if (booksStatus === 'loading') {
+    content = <h2>Loading Books...</h2>;
+  } else if (booksStatus === 'succeeded') {
+    content = books.map((book) => (
```

Comment:
- [ ] Kindly make sure when the user adds a new book, the app display the updated list of all books from the redux store. Your has a small bug of not mapping through the books from the redux store after adding a new book because you forget to check the `fullfilled` status which is being updated after adding a book instead of the `succeed` status as it is the updated one after removing a book.

```suggestion
  } else if (booksStatus === 'succeeded' || 'fullfilled') {
    content = books.map((book) => (
```

 - ### YOUR APP BEHAVIOUR 
 
  ![chrome-capture-2023-6-22](https://github.com/PedroH-Neves/book-store-react/assets/87197412/cdc3f856-92ac-43db-8109-7c7221e19d19)